python/src/scipp/compat/mantid.py

```diff
@@ -266,14 +269,22 @@ def init_pos(ws, source_pos, sample_pos):
             if spec.hasDetectors:
                 definition = spec_info.getSpectrumDefinition(i)
                 n_dets = len(definition)
+                quats = []
+                bboxes = []
                 for j in range(n_dets):
                     det_idx = definition[j][0]
                     p = det_info.position(det_idx)
+                    r = det_info.rotation(det_idx)
+                    s = comp_info.shape(det_idx)
                     spectrum_values[idx] = i
                     x_values[idx] = p.X()
                     y_values[idx] = p.Y()
                     z_values[idx] = p.Z()
                     idx += 1
+                    quats.append([r.real(), r.imagI(), r.imagJ(), r.imagK()])
+                    bboxes.append(s.getBoundingBox().width())
```

Comment:
How well does this perform? I appreciate that you do have to do something like this as you cannot know what the detector-spectra mapping sizes are exactly.

The alternative is to handle this with `groupby`. Which would probably be better given we already store the mapping information in `spectrum_values`

Reply:
I think there are probably many places that the instrument view code can be optimized.
I'll open an issue on this. Some profiling will definitely be needed.

Reply:
Can't we just use the rotations from `SpectrumInfo` and let Mantid handle the averaging?

Reply:
I don’t think the spectrumInfo gives you the rotations, unfortunately

Reply:
I don't think it would be very complex to add to Mantid. On the c++ side this is all that's required for the position averaging https://github.com/mantidproject/mantid/blob/master/Framework/API/src/SpectrumInfo.cpp#L119-L124, of course python bindings in Mantid would also need to be updated.